microprofile/graphql/server/src/main/java/io/helidon/microprofile/graphql/server/JandexUtils.java

```diff
@@ -0,0 +1,195 @@
+
+    /**
+     * Property to override the default index file. (Normally used for functional tests)
+     */
+    public static final String PROP_INDEX_FILE = "io.helidon.microprofile.graphql.indexfile";
+
+    /**
+     * The {@link Set} of loaded indexes.
+     */
+    private Set<Index> setIndexes = new HashSet<>();
+
+    /**
+     * The file used to load the index.
+     */
+    private String indexFile;
+
+    /**
+     * Construct an instance of the utilities class..
+     */
+    private JandexUtils() {
+        indexFile = System.getProperty(PROP_INDEX_FILE, DEFAULT_INDEX_FILE);
+    }
+
+    /**
+     * Create a new {@link JandexUtils}.
+     * @return a new {@link JandexUtils}
+     */
+    public static JandexUtils create() {
+         return new JandexUtils();
+    }
+
+    /**
+     * Load all the index files of the given name.
+     */
+    public void loadIndexes() {
+        try {
+            List<URL> listUrls = findIndexFiles(indexFile);
+
+            // loop through each URL and load the index
+            for (URL url : listUrls) {
+                try (InputStream input = url.openStream()) {
+                    setIndexes.add(new IndexReader(input).read());
+                } catch (Exception e) {
+                    LOGGER.warning("Unable to load default Jandex index file: " + url
+                                           + " : " + e.getMessage());
+                }
+            }
+        } catch (IOException ignore) {
+            // any Exception coming from getResources() or toURL() is ignored and
+            // the Map of indexes remain empty
+        }
+    }
+
+    /**
+     * Return all the Jandex index files with the given name. If the name is absolute then
+     * return the singl file.
+     *
+     * @param indexFileName  index file name
+     * @return a {@link List} of the index file names
+     *
+     * @throws IOException if any error
+     */
+    private List<URL> findIndexFiles(String indexFileName) throws IOException {
+        List<URL> result = new ArrayList<>();
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        File file = new File(indexFile);
+        if (file.isAbsolute()) {
+            result.add(file.toPath().toUri().toURL());
+            return result;
+        }
+
+        Enumeration<URL> urls = contextClassLoader.getResources(indexFileName);
+        while (urls.hasMoreElements()) {
+            result.add(urls.nextElement());
+        }
+
+        return result;
+    }
+
+
+    /**
+     * Return a {@link Collection} of {@link Class}es which are implementors of a given class/interface.
+     *
+     * @param clazz           {@link Class} to check for implementors
+     * @param includeAbstract indicates if abstract classes should be included
+     * @return a {@link Collection} of {@link Class}es
+     */
+    public Collection<Class<?>> getKnownImplementors(String clazz, boolean includeAbstract) {
```

Comment:
There is one capability that we need Jandex for and that is to find the annotation in the following situation.

public method(List<List<@NonNull String>> param).
I don't believe CDI can answer this, but please provide and example of how it can if i'm no correct. @tjquinno

Reply:
Actually i mis-read the above comment. Can you provide an example of how CDI can get known implementors?  @tjquinno

Reply:
After a DM with Laird, it seems that this is not so easy using CDI-only APIs. So it's probably best to keep the code as-is.